xbmc/addons/AddonInstaller.cpp

```diff
@@ -307,8 +307,10 @@ bool CAddonInstaller::CheckDependencies(const AddonPtr &addon,
                                         std::vector<std::string>& preDeps, CAddonDatabase &database,
                                         std::pair<std::string, std::string> &failedDep)
 {
-  if (addon == NULL)
-    return true; // a NULL addon has no dependencies
+  // repositories and nullptr addons have no dependencies
+
+  if (addon->HasType(ADDON_REPOSITORY) || addon == nullptr)
+    return true;
```

Comment:
Can you explain why this mean repo add-ons cannot have any dependencies?

Reply:
Or would it be better to show a dialog to the user that a repo addon will not be installed as it has dependencies rather than silently omitting them.

Reply:
yes, i think a warning dialog or maybe a log entry would be appropiate. question is if we should deny the whole repo then or just leave the dependencies out?

Reply:
Maybe just a toast if it fails. Similar to any other failed addon install.